### PR TITLE
fix: API /teams/:id/publish

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Fixes API build failure introduced after #9291

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [x] Locally or on preview, check that build succeeds
- [ ] Test the [publish endpoint](https://www.postman.com/calcom/workspace/cal-com-workspace/request/20666831-c71e7234-fdc0-4f94-847a-a36bcb1434eb) against a valid api key
  - Locally, run `yarn dev:api`
  - Login to webapp and create an api key localhost:3000/settings/developer/api-keys
  - Copy the api key into the postman environment variable
  - On Postman select the local environment and overwrite the api_key with your own: 
image
image

  - Test the endpoint with valid body data as in the postman example

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/api/lib/helpers/verifyApiKey.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export const dateNotInPast = function (date: Date) {
 export const verifyApiKey: NextMiddleware = async (req, res, next) => {
   const { prisma, isCustomPrisma, isAdmin } = req;
   const hasValidLicense = await checkLicense(prisma);
-  if (!hasValidLicense)
+  if (!hasValidLicense && process.env.NODE_ENV !== "development")
```
Comment (zomars) [inline apps/api/lib/helpers/verifyApiKey.ts:20]: So we can test API on development

File: apps/api/pages/api/teams/[teamId]/publish.ts [hunk 2]
```diff
@@ -2,29 +2,36 @@ import type { NextApiRequest, NextApiResponse } from "next";
...
+    };
   }
+  /** @see https://trpc.io/docs/server-side-calls */
+  const ctx = await createContext({ req, res }, sessionGetter);
   try {
-    const { teamId } = schemaQueryTeamId.parse(req.query);
-    return await publishHandler({ input: { teamId }, ctx: { ...ctx, user } });
+    const caller = viewerTeamsRouter.createCaller(ctx);
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return await caller.publish(req.query as any /* Let tRPC handle this */);
```
Comment (zomars) [inline apps/api/pages/api/teams/[teamId]/publish.ts:34]: Usted the same approach we have on slots and confirm link endpoints.

File: packages/trpc/server/routers/viewer/teams/publish.schema.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { z } from "zod";
 
 export const ZPublishInputSchema = z.object({
-  teamId: z.number(),
+  teamId: z.coerce.number(),
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/teams/publish.schema.ts:4]: By default api paths are strings, so we update to account into this.

**Review by zomars — COMMENTED** [review]
Ready for review

**Review by emrysal — APPROVED** [review]
lgtm

**Pull request merged**